packages/web-components/src/components/masthead/megamenu-tabs.ts

```diff
@@ -35,6 +35,14 @@ class DDSMegaMenuTabs extends BXTabs {
     return `${ddsPrefix}-megamenu-tab[selected]`;
   }
 
+  /**
+   * The name of the custom event fired before a tab is selected upon a user gesture.
+   * Cancellation of this event stops changing the user-initiated selection.
+   */
+  static get eventBeforeSelect() {
+    return `${ddsPrefix}-megamenu-tabs-beingselected`;
```

Comment:
We're defining this same event name in masthead-composite.ts:1554. Should we really have the same event title defined on two different components? Are they both in use?

Reply:
This is the library's existing pattern of emitting and listening for custom events. Here's another example used in the Masthead:
- Emitting component: https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/feat/masthead-v2-dev/packages/web-components/src/components/masthead/megamenu-top-nav-menu.ts#L94
- Listening component: https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/feat/masthead-v2-dev/packages/web-components/src/components/masthead/masthead-composite.ts#L1500-L1502

Now that you call attention to it, though, I don't like the pattern very much. We have to use the same magic string twice, whereas we could just as easily define it as a static property on the emitting class, then access that property on the listening class. I'll update.